packages/tasks_panel/test/multi-step-select.spec.ts

```diff
@@ -93,38 +105,74 @@ describe("multi-step-selection scope", () => {
     );
   });
 
+  it("grabTasksByGroup - `build` group required", async () => {
+    mockFioriE2eCinfig.expects("getConfigDeployPickItems").never();
+    expect(await __internal.grabTasksByGroup(tasks, roots[0], "BuIld")).to.be.deep.equal([
+      { label: taskContributed1.__intent, kind: testVscode.QuickPickItemKind.Separator },
+      Object.assign(
```

Comment:
consider spread operator instead of Object.assign + cloneDeep

```javascript
      {
        ...taskContributed1,
        ...{ description: taskContributed1.type },
        ...{ detail: taskContributed1.description }
      },
```

But we don't need multiple spread operators, so we can improve this to.

```javascript
      {
        ...taskContributed1,
        description: taskContributed1.type,
        detail: taskContributed1.description
      },
```

Note that `...` is not equivalent to `deepClone`, it is a shallowClone so maybe you still need `deepClone`
Depending on the nesting level of `taskContributed1`